#!/bin/sh -e
##
## 91_debian_alsaconf_noclear.dpatch by Thomas Hood
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't clear the screen. (Let the user see error messages.)

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad alsa-utils-1.0.8/alsaconf/alsaconf.in /tmp/dpep.5AMwxh/alsa-utils-1.0.8/alsaconf/alsaconf.in
--- alsa-utils-1.0.8/alsaconf/alsaconf.in	2005-02-18 21:40:46.000000000 +0100
+++ /tmp/dpep.5AMwxh/alsa-utils-1.0.8/alsaconf/alsaconf.in	2005-02-18 21:40:46.000000000 +0100
@@ -366,10 +366,19 @@
   $DIALOG --msgbox "$msg" 17 60 || acex 0
 }
 
+clear() {
+    echo
+}
+
 # Exit function
 acex() {
   cleanup
-  clear
+  if [ "$1" = 0 ] ; then
+      clear
+  else
+      # Don't clear error messages
+      echo
+  fi
   exit $1
 }
 
